rust/arrow/src/compute/kernels/aggregate.rs

```diff
@@ -135,6 +137,40 @@ where
     Some(n)
 }
 
+/// Helper function to perform min/max lambda function on values from a `BooleanArray`
+fn min_max_boolean<F>(array: &BooleanArray, cmp: F) -> Option<bool>
+where
+    F: Fn(bool, bool) -> bool,
+{
+    let null_count = array.null_count();
+
+    // Includes case array.len() == 0
+    if null_count == array.len() {
+        return None;
+    }
+
+    let m0: Option<bool> = array.iter().next().unwrap();
+
+    array.iter().fold(m0, |max, item| match (max, item) {
```

Comment:
Would it make sense to return early here when seeing the highest or lowest value?

Reply:
Good point. This is indeed simpler given the finite number of choices. Something like `.any(item)` would short-circuit, I think.

Reply:
Addressed in https://github.com/apache/arrow/pull/8946/commits/075277c7ae64c8d1aacf7a07c7cbf501c945570b -- I am actually quite happy with how the code looks now given this suggestion. 👍